ocs_ci/ocs/platform_nodes.py

```diff
@@ -2420,3 +2420,62 @@ def restart_nodes_by_stop_and_start_teardown(self):
                 cmd = f"ibmcloud ks worker reboot --cluster {cluster_id} --worker {not_ready_node} -f"
                 out = run_cmd(cmd)
                 logger.info(f"Node restart command output: {out}")
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        Create node on IBM Cloud.
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created
+            num_nodes (int): Number of node instances to be created
+
+        Returns:
+           list: of IBMCloudNode objects
+
+        """
+        logger.info("creating new node")
+        login()
+
+        worker_nodes = get_nodes(node_type="worker")
+        provider_id = worker_nodes[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+
+        cmd = f"ibmcloud ks workers --cluster {cluster_id} --output json"
+        out = run_cmd(cmd)
+        worker_nodes = json.loads(out)
+
+        no_of_nodes = len(worker_nodes) + int(num_nodes)
+        logger.info(f"number of nodes going to be add are : {num_nodes}")
+
+        cmd = (
+            f"ibmcloud ks worker-pool resize --cluster {cluster_id} --worker-pool default"
+            f"  --size-per-zone {no_of_nodes}"
+        )
+        run_cmd(cmd)
+
+        logger.info("Waiting for 600 seconds for node to ready")
+        time.sleep(600)
+
+        worker_nodes = get_nodes(node_type="worker")
+        worker_nodes_not_ready = []
+        for worker_node in worker_nodes:
+            node_id = worker_node.get()["metadata"]["labels"][
+                "ibm-cloud.kubernetes.io/worker-id"
+            ]
+            logger.info(f"{node_id} status is : {worker_node.status()}")
+            if worker_node.status() != "Ready":
+                worker_nodes_not_ready.append(node_id)
+
+        if len(worker_nodes_not_ready) > 0:
+            time.sleep(600)
```

Comment:
again - here you are waiting 10 mins - and in the case those nodes are still in not ready state you don't even know it and continue. If this is still the case you should raise the exception here that nodes are not in ready state.

Reply:
Done